src/app/space/settings/areas/create-area-dialog/create-area-dialog.component.html

```diff
@@ -11,15 +11,20 @@ <h3>
           <span class="pficon pficon-error-circle-o"></span>
           <b>Unique Name</b> A unique name is required to add a new area.
         </div>
-        <div [hidden]="!(inputModel.hasError('required')) || inputModel.pristine" class="alert alert-danger">
+        <div [hidden]="!errors.exceedLengthFailure" class="alert alert-danger">
+          <span class="pficon pficon-error-circle-o"></span>
+          <b>Name Too Long</b> A name must not exceed 63 characters.
+        </div>
+        <div [hidden]="!errors.emptyNameFailure" class="alert alert-danger">
           <span class="pficon pficon-error-circle-o"></span>
           <b>Missing name</b> A name is required to add a new area.
         </div>
-        <div class="form-group" [ngClass]="{'has-error': errors.uniqueValidationFailure || !inputModel.pristine && !inputModel.valid}">
+        <div class="form-group" [ngClass]="{'has-error': errors.uniqueValidationFailure || errors.exceedLengthFailure || errors.emptyNameFailure || !inputModel.pristine && !inputModel.valid}">
```

Comment:
Could you add a function, either to the AreaError interface/impl or to the dialog component itself, so that the template doesn't need to `||` over all of the error type properties like this?

Reply:
Or maybe rather than a series of error flags, this information can be encoded with an enum type instead? This would make the most sense if invalidations are mostly due to disjoint reasons though (ie "too long" and "empty" can never both be the case, and "too long" or "empty" should block creation and imply that "non-unique" could not also be the case as well).

Reply:
Yeah that sounds cleaner, I've added an enum called 'AreaCreationStatus' that has an 'ok' and 'error' state, and it's now used in the html for checking errors instead.